File: python/src/blind150/arrays/TwoSum.py
from typing import List
from src.utils.logging_util import get_logger

logger = get_logger(__name__)

class TwoSum:
    def compute(self, nums: List[int], target: int) -> List[int]:
        targetMap = {}

        for i, n in enumerate(nums):
            diff = target - n
            logger.debug(f"Checking number: {n}, diff: {diff}, targetMap: {targetMap}")
            if diff in targetMap:
                logger.info(f"Found pair: {n} + {diff} = {target}, at indices {targetMap[diff]} and {i}")
                return [targetMap[diff], i]
            targetMap[n] = i
        logger.warning(f"No pair found for target {target}")
        return [-1, -1]

# Driver code
if __name__ == "__main__":
    two_sum = TwoSum()

    nums = [0, 6, 2, 7, 11, 15]
    target = 9
    logger.info(f"Input array: {nums}, target: {target}")
    result = two_sum.compute(nums, target)

    if result != [-1, -1]:
        logger.info(f"Indices of numbers that add up to {target}: {result}")
    else:
        logger.warning("No valid pair found.")

File: java/src/main/java/dsa/blind150/arrays/TwoSum.java
package dsa.blind150.arrays;

import dsa.utils.LoggerUtil;
import org.slf4j.Logger;

import java.util.HashMap;

public class TwoSum {
    private static final Logger logger = LoggerUtil.getLogger();
    public static int[] find(int[] array, int target) {
        HashMap<Integer, Integer> hashMap = new HashMap<>();

        for (int i = 0; i < array.length; i++) {
            int num = array[i];
            int targetNum = target - num;

            if (hashMap.containsKey(targetNum)) {
                logger.info("Found pair: {} + {} = {}", num, targetNum, target);
                return new int[]{hashMap.get(targetNum), i};
            }

            hashMap.put(num, i);
        }

        logger.warn("No pair found that adds up to {}", target);
        return new int[]{-1, -1};
    }

    public static void main(String[] args) {
        int[] nums = {0, 6, 2, 7, 11, 15};
        int target = 9;

        int[] result = find(nums, target);

        if (result[0] != -1 && result[1] != -1) {
            logger.info("Indices found at: {}, {}", result[0], result[1]);
        } else {
            logger.warn("No valid pair found.");
        }
    }
}